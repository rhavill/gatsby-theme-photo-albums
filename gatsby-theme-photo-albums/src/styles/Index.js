/** @jsx jsx */
import { jsx } from 'theme-ui'
import PropTypes from 'prop-types'

const Index = ({children}) => 
  <div sx={{variant: 'indexPage'}}>
    {children}
  </div>

Index.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default Index