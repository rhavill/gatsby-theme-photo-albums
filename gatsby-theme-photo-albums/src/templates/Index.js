import React from 'react'
import PropTypes from 'prop-types'
import {graphql} from 'gatsby'
import compose from 'ramda/src/compose'
import map from 'ramda/src/map'
import merge from 'ramda/src/merge'
import prop from 'ramda/src/prop'
import Layout from '../components/layout/Layout'
import Folders from '../components/folders/Folders'
import Thumbnails from '../components/thumbnails/Thumbnails'
import Pager from '../components/pagination/Pager'
import {addUrlProps} from '../util/files-folders'
import {removePathPrefix} from '../util/url-text'

const Index = ({data, location, pageContext}) => {
  const {baseUrl, currentPage, numPages} = pageContext
  const pathPrefix = data.site.pathPrefix
  const path = compose(
    removePathPrefix(pathPrefix), decodeURI
  )(location.pathname)
  const folders = compose(
    map(prop('url')),
    addUrlProps(baseUrl))(data.folders.nodes)
  const files = compose(
    map(file => merge(file, {imageData: file.childImageSharp.fixed})),
    addUrlProps(baseUrl))(data.photos.nodes)
  const folderIcon = data.folderIcon.childImageSharp.fixed

  return (
    <Layout path={path}>
      <div className="listing-page" data-testid={path} >
        <section>
          <Folders path={path} folders={folders} icon={folderIcon} />
          <Thumbnails files={files} currentPage={currentPage} />
        </section>
      </div>
      <Pager path={path} currentPage={currentPage} numPages={numPages} />
      <br/>
    </Layout>
  )
}

export const query = graphql`
  query indexQuery($skip: Int!, $limit: Int!, $relativeDirectory: String!) {
    site {
      pathPrefix
    }
    photos: allFile(
      filter: {
        sourceInstanceName: {eq: "gtpaPhotos"}, 
        relativeDirectory: {eq: $relativeDirectory}
      }, 
      sort: {fields: relativePath} 
      limit: $limit 
      skip: $skip) {
      nodes {
        relativePath
        childImageSharp {
          fixed(width: 250, height: 250, cropFocus: CENTER) {
            ...GatsbyImageSharpFixed
          }
        }      
      }
    }
    folderIcon: file(sourceInstanceName: {eq: "gtpaFolderIcon"}) {
      childImageSharp {
        fixed(width: 250, height: 250) {
          ...GatsbyImageSharpFixed
        }
      }
    }
    folders: allDirectory(
      filter: {
        sourceInstanceName: {eq: "gtpaPhotos"},
        relativeDirectory: {eq: $relativeDirectory}
      }, 
      sort: {fields: relativePath}) {
      nodes {
        relativePath
      }
    }
  }
`

Index.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  }).isRequired,
  data: PropTypes.shape({
    site: PropTypes.shape({
      pathPrefix: PropTypes.string.isRequired,
    }),
    folderIcon: PropTypes.shape({
      childImageSharp: PropTypes.shape({
        fixed: PropTypes.object.isRequired
      }).isRequired       
    }).isRequired,
    folders: PropTypes.shape({
      nodes: PropTypes.arrayOf(
        PropTypes.shape({
          relativePath: PropTypes.string.isRequired,
        }).isRequired   
      ).isRequired
    }).isRequired,
    photos: PropTypes.shape({
      nodes: PropTypes.arrayOf(
        PropTypes.shape({
          relativePath: PropTypes.string.isRequired,
          childImageSharp: PropTypes.shape({
            fixed: PropTypes.object.isRequired,
          })
        }).isRequired   
      ).isRequired
    }).isRequired,
  }).isRequired,
  pageContext: PropTypes.shape({
    currentPage: PropTypes.number.isRequired,
    numPages: PropTypes.number.isRequired,
    relativeDirectory: PropTypes.string.isRequired,
    baseUrl: PropTypes.string.isRequired,
  }).isRequired
}

export default Index