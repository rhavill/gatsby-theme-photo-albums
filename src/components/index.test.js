import React from 'react'
import {StaticQuery} from 'gatsby'
import renderer from 'react-test-renderer'
import {cleanup, render} from '@testing-library/react'
import Index from '../pages/index'
import queryResults from '../test-data/index-page-graphql-results'

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup)
beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render(queryResults.data)
  )
})

describe('Index', () => {
  it('renders correctly', () => {
    const pageContext = {numPages: 1, currentPage: 1, regexFilter: '/^[^/]+$/'}
    const tree = renderer
      .create(<Index location={{pathname: '/'}} data={queryResults.data} pageContext={pageContext}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('passes decoded urls to child components', () => {
    const pageContext = {numPages: 1, currentPage: 1, regexFilter: '/^san-sebastián[^/]+$/'}
    const {getAllByTestId} = render(
      <Index location={{pathname: '/san-sebasti%c3%a1n'}} data={queryResults.data} pageContext={pageContext}/>
    )
    expect(getAllByTestId('/san-sebastián').length).toBe(1)
  })
})