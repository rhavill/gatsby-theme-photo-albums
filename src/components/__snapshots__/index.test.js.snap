// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
<div
  className="wrapper"
>
  <header>
    <nav
      className="container"
    >
      <a
        href="/"
      >
        Home
      </a>
    </nav>
  </header>
  <div
    className="container page"
  >
    <div
      className="listing-page"
    >
      <section>
        <article
          className="folder"
          role="folder"
        >
          <a
            href="/2019-puerto-rico"
          >
            <div
              className=" gatsby-image-wrapper"
              style={
                Object {
                  "display": "inline-block",
                  "height": 250,
                  "overflow": "hidden",
                  "position": "relative",
                  "width": 250,
                }
              }
            >
              <picture>
                <source
                  srcSet="/static/b75d8bd08d0e30f3925cc6187ddc8b84/f17d3/folder.png 1x"
                />
                <img
                  alt="2019 Puerto Rico"
                  height={250}
                  loading="lazy"
                  onLoad={[Function]}
                  src="/static/b75d8bd08d0e30f3925cc6187ddc8b84/f17d3/folder.png"
                  srcSet="/static/b75d8bd08d0e30f3925cc6187ddc8b84/f17d3/folder.png 1x"
                  style={
                    Object {
                      "height": "100%",
                      "left": 0,
                      "objectFit": "cover",
                      "objectPosition": "center",
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transition": "opacity 500ms",
                      "width": "100%",
                    }
                  }
                  width={250}
                />
              </picture>
              <noscript
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<picture><source srcset=\\"/static/b75d8bd08d0e30f3925cc6187ddc8b84/f17d3/folder.png 1x\\" /><img loading=\\"lazy\\" width=\\"250\\" height=\\"250\\" srcset=\\"/static/b75d8bd08d0e30f3925cc6187ddc8b84/f17d3/folder.png 1x\\" src=\\"/static/b75d8bd08d0e30f3925cc6187ddc8b84/f17d3/folder.png\\" alt=\\"2019 Puerto Rico\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                  }
                }
              />
            </div>
            <div
              className="folder-title"
            >
              2019 Puerto Rico
            </div>
          </a>
        </article>
        <article
          className="file"
          role="file"
        >
          <a
            href="/IMG_20190810_110416612-small.jpg"
          >
            <div
              className=" gatsby-image-wrapper"
              style={
                Object {
                  "display": "inline-block",
                  "height": 250,
                  "overflow": "hidden",
                  "position": "relative",
                  "width": 250,
                }
              }
            >
              <picture>
                <source
                  srcSet="/static/bd5e73a4e4ef1e6c999f05a63948fb15/284df/IMG_20190814_113735817-small.jpg 1x"
                />
                <img
                  alt="Img_20190810_110416612 Small"
                  height={250}
                  loading="lazy"
                  onLoad={[Function]}
                  src="/static/bd5e73a4e4ef1e6c999f05a63948fb15/284df/IMG_20190814_113735817-small.jpg"
                  srcSet="/static/bd5e73a4e4ef1e6c999f05a63948fb15/284df/IMG_20190814_113735817-small.jpg 1x"
                  style={
                    Object {
                      "height": "100%",
                      "left": 0,
                      "objectFit": "cover",
                      "objectPosition": "center",
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transition": "opacity 500ms",
                      "width": "100%",
                    }
                  }
                  width={250}
                />
              </picture>
              <noscript
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<picture><source srcset=\\"/static/bd5e73a4e4ef1e6c999f05a63948fb15/284df/IMG_20190814_113735817-small.jpg 1x\\" /><img loading=\\"lazy\\" width=\\"250\\" height=\\"250\\" srcset=\\"/static/bd5e73a4e4ef1e6c999f05a63948fb15/284df/IMG_20190814_113735817-small.jpg 1x\\" src=\\"/static/bd5e73a4e4ef1e6c999f05a63948fb15/284df/IMG_20190814_113735817-small.jpg\\" alt=\\"Img_20190810_110416612 Small\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                  }
                }
              />
            </div>
          </a>
        </article>
      </section>
    </div>
    <div
      className="pager"
    />
    <br />
  </div>
</div>
`;
